### PR TITLE
Update logging to use key value pairs

### PR DESCRIPTION
**Pull request by ryangrundy7**
# Motivation and Context
This PR updates all the logging to use key value pairs from the Godaddy logging package

# What has changed
Updates all logging to use key value pairs.

# How to test?
Run tests
Make sure any log lines haven't been missed

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/sample/SampleSvcApplication.java [hunk 1]
```diff
@@ -58,6 +59,13 @@ public static void main(final String[] args) {
     SpringApplication.run(SampleSvcApplication.class, args);
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/sample/SampleSvcApplication.java]: Need to remove `LoggingConfigs.setCurrent(LoggingConfigs.getCurrent().useJson());`

File: src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java [hunk 2]
```diff
@@ -77,9 +77,8 @@ public SampleEndpoint(SampleService sampleService, MapperFacade mapperFacade) {
               collectionExerciseJobCreationRequestDTO,
       BindingResult bindingResult)
       throws CTPException, InvalidRequestException {
-    log.debug(
-        "Entering createCollectionExerciseJob with requestObject {}",
-        collectionExerciseJobCreationRequestDTO);
+    log.with("collection_exercise_job_creation_request", collectionExerciseJobCreationRequestDTO)
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: No need to log out the whole DTO object

File: src/main/java/uk/gov/ons/ctp/response/sample/message/SampleUnitPublisher.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
    * @param sampleUnit to be sent
    */
   public void send(SampleUnit sampleUnit) {
-    log.debug("send to queue sampleDelivery {}", sampleUnit);
+    log.with("sampleUnit", sampleUnit).debug("send to queue sampleDelivery");
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/sample/message/SampleUnitPublisher.java]: sample_unit

File: src/main/java/uk/gov/ons/ctp/response/sample/scheduled/distribution/SampleUnitDistributor.java [hunk 4]
```diff
@@ -88,8 +88,8 @@ private void processJob(CollectionExerciseJob job) {
         sampleUnitSender.sendSampleUnit(msu);
       } catch (CTPException e) {
         hasErrors = true;
-        log.error(
-            "Failed to send a sample unit to queue and update state with ID: {}", msu.getId(), e);
+        log.with("mapped_sample_unit_id", msu.getId())
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/sample/scheduled/distribution/SampleUnitDistributor.java]: sample_unit_id?

File: src/main/java/uk/gov/ons/ctp/response/sample/service/PartySvcClientService.java [hunk 5]
```diff
@@ -58,11 +58,12 @@ public PartyDTO postParty(final PartyCreationRequestDTO partyCreationRequestDTO)
       try {
         party = objectMapper.readValue(responseBody, PartyDTO.class);
       } catch (IOException e) {
-        String msg = String.format("cause = %s - message = %s", e.getCause(), e.getMessage());
-        log.error(msg);
+        log.with("cause", e.getCause())
+            .with("message", e.getMessage())
+            .error("Unable to read party response");
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/sample/service/PartySvcClientService.java]: I done the same but `.error("Unable to read party response", e)` makes more sense as @NickGrantONS suggested

File: src/main/java/uk/gov/ons/ctp/response/sample/service/PartySvcClientService.java [hunk 6]
```diff
@@ -58,11 +58,10 @@ public PartyDTO postParty(final PartyCreationRequestDTO partyCreationRequestDTO)
       try {
         party = objectMapper.readValue(responseBody, PartyDTO.class);
       } catch (IOException e) {
-        String msg = String.format("cause = %s - message = %s", e.getCause(), e.getMessage());
-        log.error(msg);
+        log.error("Unable to read party response", e);
       }
     }
-    log.debug("PARTY GOTTEN: {}", party);
+    log.with("party", party).debug("PARTY GOTTEN");
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/sample/service/PartySvcClientService.java]: `"Successfully retrieved party"`?

File: src/main/java/uk/gov/ons/ctp/response/sample/message/SampleUnitPublisher.java [hunk 7]
```diff
@@ -25,7 +25,7 @@
    * @param sampleUnit to be sent
    */
   public void send(SampleUnit sampleUnit) {
-    log.with("sampleUnit", sampleUnit).debug("send to queue sampleDelivery");
+    log.with("sample_Unit", sampleUnit).debug("send to queue sampleDelivery");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/sample/message/SampleUnitPublisher.java]: lower case U

File: src/main/java/uk/gov/ons/ctp/response/sample/service/PartySvcClientService.java [hunk 8]
```diff
@@ -58,12 +58,10 @@ public PartyDTO postParty(final PartyCreationRequestDTO partyCreationRequestDTO)
...
         party = objectMapper.readValue(responseBody, PartyDTO.class);
       } catch (IOException e) {
-        log.with("cause", e.getCause())
-            .with("message", e.getMessage())
-            .error("Unable to read party response");
+        log.error("Unable to read party response", e);
       }
     }
-    log.with("party", party).debug("PARTY GOTTEN");
+    log.with("Successfully retrieved party", party).debug("PARTY GOTTEN");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/sample/service/PartySvcClientService.java]: replace "PARTY GOTTEN" with "Successfullly..." and set the key back to be party

**Pull request merged**